### PR TITLE
Downloads: Improve error message url is invalid

### PR DESCRIPTION
**Pull request by autoantwort**
Fixes https://github.com/microsoft/vcpkg/issues/38877

### CODE REVIEW

File: src/vcpkg/base/downloads.cpp [hunk 1]
```diff
@@ -764,9 +764,11 @@ namespace vcpkg
         if (headers.size() == 0)
         {
             auto split_uri = split_uri_view(url).value_or_exit(VCPKG_LINE_INFO);
-            auto authority = split_uri.authority.value_or_exit(VCPKG_LINE_INFO).substr(2);
             if (split_uri.scheme == "https" || split_uri.scheme == "http")
             {
+                Checks::msg_check_exit(
+                    VCPKG_LINE_INFO, split_uri.authority.has_value(), msgInvalidUri, msg::value = url);
+                auto authority = split_uri.authority.value_or_exit(VCPKG_LINE_INFO).substr(2);
```
Comment (BillyONeal) [inline src/vcpkg/base/downloads.cpp:771]:
```suggestion
                auto maybe_authority = split_uri.authority.get();
                if (!maybe_authority) {
                Checks::msg_exit_with_error(
                    VCPKG_LINE_INFO, split_uri.authority.has_value(), msgInvalidUri, msg::value = url);
                }

                auto authority = maybe_authority->substr(2);
```

I find the substr(2) confusing but...

**Review by BillyONeal — COMMENTED** [review]
Can you add a test? I think this one should be unit testable...

**Comment by autoantwort** [comment]
> Can you add a test? I think this one should be unit testable...

There are tests for `split_uri_view`, but I haven't changed that function. And this functions calls curl, so I don't think that this function (`try_download_file`) is unit testable.

**Comment by BillyONeal** [comment]
Ah, I see, when I first read this I read the `substr(2)` as new parsing code that should be tested but I see now you only moved it down from above.

**Review by BillyONeal — APPROVED** [review]

**Pull request merged**